Components/RTShaderSystem/src/OgreShaderGenerator.cpp

```diff
@@ -965,13 +965,9 @@ bool ShaderGenerator::cloneShaderBasedTechniques(Material& srcMat, Material& dst
                         RenderState* srcRenderState = t->getRenderState(pi);
                         RenderState* dstRenderState = getRenderState(srcToTechniqueScheme, dstMat, pi);
 
-                        const SubRenderStateList& srcSubRenderState =
-                            srcRenderState->getSubRenderStates();
-
-                        SubRenderStateList::const_iterator itSubState = srcSubRenderState.begin(), itSubStateEnd = srcSubRenderState.end();
-                        for(;itSubState != itSubStateEnd ; ++itSubState)
+                        for(const auto& s : srcRenderState->getSubRenderStates())
```

Comment:
```suggestion
                        for(SubRenderState* srcSubState : srcRenderState->getSubRenderStates())
```

Reply:
I did this 
for (SubRenderState* s : srcRenderState->getSubRenderStates())